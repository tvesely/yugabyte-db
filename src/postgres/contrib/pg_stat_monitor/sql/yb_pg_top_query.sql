CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;
SELECT pg_stat_monitor_reset();
CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS
$$
BEGIN
	return (select $1 + $2);
END; $$ language plpgsql;

CREATE OR REPLACE function add2(int, int) RETURNS int as
$$
BEGIN
	return add($1,$2);
END;
$$ language plpgsql;

SELECT add2(1,2);
-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: Top query has run to run variability in YB
-- SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C";
SELECT query FROM pg_stat_monitor ORDER BY query COLLATE "C";
SELECT pg_stat_monitor_reset();
DROP EXTENSION pg_stat_monitor;
