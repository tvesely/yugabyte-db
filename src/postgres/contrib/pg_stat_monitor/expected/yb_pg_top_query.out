CREATE EXTENSION IF NOT EXISTS pg_stat_monitor;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS
$$
BEGIN
	return (select $1 + $2);
END; $$ language plpgsql;
CREATE OR REPLACE function add2(int, int) RETURNS int as
$$
BEGIN
	return add($1,$2);
END;
$$ language plpgsql;
SELECT add2(1,2);
 add2
------
    3
(1 row)

-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: Top query has run to run variability in YB
-- SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C";
SELECT query FROM pg_stat_monitor ORDER BY query COLLATE "C";
                             query
---------------------------------------------------------------
 CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS  +
 $$                                                           +
 BEGIN                                                        +
         return (select $1 + $2);                             +
 END; $$ language plpgsql;
 CREATE OR REPLACE function add2(int, int) RETURNS int as     +
 $$                                                           +
 BEGIN                                                        +
         return add($1,$2);                                   +
 END;                                                         +
 $$ language plpgsql;
 SELECT (select $1 + $2)
 SELECT add2($1,$2)
 SELECT pg_stat_monitor_reset();
 SELECT query FROM pg_stat_monitor ORDER BY query COLLATE "C";
(6 rows)

SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

DROP EXTENSION pg_stat_monitor;
